### ISSUE TITLE
Core data update issue since 1.13.1

### ISSUE THREAD

**Issue by bonebox** [hyperoslo/Sync#272]
Hey @3lvis,

Since 1.13.1, Sync throws an error when I attempt to sync the same entity a second time.  The initial (insert) works, but a following sync (update) causes the crash.  I also tried 1.13.2 with the same problem.  Reverting to 1.13.0 resolves the problem, so it appears to be something introduced in 1.13.1.  I've included the relevant info below. As you can see, there is no `uri` attribute for `ModelProperty`, so the error is odd.  Let me know if you need anything else!

```
*** Terminating app due to uncaught exception 'NSUnknownKeyException', reason: '[<ModelProperty 0x7fdbe262ce60> valueForUndefinedKey:]: the entity ModelProperty is not key value coding-compliant for the key "uri".'
*** First throw call stack:
(
	0   CoreFoundation                      0x0000000106d21d85 __exceptionPreprocess + 165
	1   libobjc.A.dylib                     0x0000000107560deb objc_exception_throw + 48
	2   CoreFoundation                      0x0000000106d219c9 -[NSException raise] + 9
	3   CoreData                            0x0000000110570851 -[NSManagedObject valueForUndefinedKey:] + 289
	4   Foundation                          0x0000000107144497 -[NSSet(NSKeyValueCoding) valueForKey:] + 389
	5   Sync                                0x000000010529da36 _TFE4SyncCSo15NSManagedObject23sync_toManyRelationshipfTCSo25NSRelationshipDescription10dictionaryGVs10DictionarySSPs9AnyObject__6parentGSqS0__18parentRelationshipGSqS1__9dataStackC9DATAStack9DATAStack10operationsVC10DATAFilter10DATAFilter9Operation_T_ + 9350
	6   Sync                                0x000000010529a617 _TFE4SyncCSo15NSManagedObject23sync_fillWithDictionaryfTGVs10DictionarySSPs9AnyObject__6parentGSqS0__18parentRelationshipGSqCSo25NSRelationshipDescription_9dataStackC9DATAStack9DATAStack10operationsVC10DATAFilter10DATAFilter9Operation_T_ + 3943
	7   Sync                                0x00000001052a14a0 _TFE4SyncCSo15NSManagedObject22sync_toOneRelationshipfTCSo25NSRelationshipDescription10dictionaryGVs10DictionarySSPs9AnyObject__9dataStackC9DATAStack9DATAStack10operationsVC10DATAFilter10DATAFilter9Operation_T_ + 5376
	8   Sync                                0x000000010529b4a9 _TFE4SyncCSo15NSManagedObject23sync_fillWithDictionaryfTGVs10DictionarySSPs9AnyObject__6parentGSqS0__18parentRelationshipGSqCSo25NSRelationshipDescription_9dataStackC9DATAStack9DATAStack10operationsVC10DATAFilter10DATAFilter9Operation_T_ + 7673
	9   Sync                                0x00000001052af1b6 _TFZFC4Sync4Sync7changesFTGSaGVs10DictionarySSPs9AnyObject___13inEntityNamedSS9predicateGSqCSo11NSPredicate_6parentGSqCSo15NSManagedObject_18parentRelationshipGSqCSo25NSRelationshipDescription_9inContextCSo22NSManagedObjectContext9dataStackC9DATAStack9DATAStack10operationsVC10DATAFilter10DATAFilter9Operation10completionGSqFT5errorGSqCSo7NSError__T___T_U0_FTGS1_SSPS2___S4__T_ + 198
	10  DATAFilter                          0x0000000104a38b5d _TZFC10DATAFilter10DATAFilter7changesfTGSaGVs10DictionarySSPs9AnyObject___13inEntityNamedSS9predicateGSqCSo11NSPredicate_10operationsVS0_9Operation15localPrimaryKeySS16remotePrimaryKeySS7contextCSo22NSManagedObjectContext8insertedFT4JSONGS1_SSPS2____T_7updatedFT4JSONGS1_SSPS2___13updatedObjectCSo15NSManagedObject_T__T_ + 8893
	11  Sync                                0x00000001052aae27 _TZFC4Sync4Sync7changesfTGSaGVs10DictionarySSPs9AnyObject___13inEntityNamedSS9predicateGSqCSo11NSPredicate_6parentGSqCSo15NSManagedObject_18parentRelationshipGSqCSo25NSRelationshipDescription_9inContextCSo22NSManagedObjectContext9dataStackC9DATAStack9DATAStack10operationsVC10DATAFilter10DATAFilter9Operation10completionGSqFT5errorGSqCSo7NSError__T___T_ + 2983
	12  Sync                                0x00000001052ae4d5 _TFZFC4Sync4Sync7changesFTGSaGVs10DictionarySSPs9AnyObject___13inEntityNamedSS9predicateGSqCSo11NSPredicate_9dataStackC9DATAStack9DATAStack10operationsVC10DATAFilter10DATAFilter9Operation10completionGSqFT5errorGSqCSo7NSError__T___T_U_FCSo22NSManagedObjectContextT_ + 469
	13  Sync                                0x00000001052a96cf _TPA__TFZFC4Sync4Sync7changesFTGSaGVs10DictionarySSPs9AnyObject___13inEntityNamedSS9predicateGSqCSo11NSPredicate_9dataStackC9DATAStack9DATAStack10operationsVC10DATAFilter10DATAFilter9Operation10completionGSqFT5errorGSqCSo7NSError__T___T_U_FCSo22NSManagedObjectContextT_ + 255
	14  DATAStack                           0x0000000104ac28d3 _TFFC9DATAStack9DATAStack29performInNewBackgroundContextFFT17backgroundContextCSo22NSManagedObjectContext_T_T_U_FT_T_ + 67
	15  DATAStack                           0x0000000104ac2917 _TTRXFo__dT__XFdCb__dT__ + 39
	16  CoreData                            0x0000000110508127 developerSubmittedBlockToNSManagedObjectContextPerform + 199
	17  libdispatch.dylib                   0x0000000107ff93eb _dispatch_client_callout + 8
	18  libdispatch.dylib                   0x0000000107fdf82c _dispatch_queue_drain + 2215
	19  libdispatch.dylib                   0x0000000107fded4d _dispatch_queue_invoke + 601
	20  libdispatch.dylib                   0x0000000107fe1996 _dispatch_root_queue_drain + 1420
	21  libdispatch.dylib                   0x0000000107fe1405 _dispatch_worker_thread3 + 111
	22  libsystem_pthread.dylib             0x0000000108336746 _pthread_wqthread + 1299
	23  libsystem_pthread.dylib             0x0000000108336221 start_wqthread + 13
)
libc++abi.dylib: terminating with uncaught exception of type NSException
```
The model:

<img width="571" alt="screen shot 2016-09-01 at 2 27 38 pm" src="https://cloud.githubusercontent.com/assets/1120504/18183623/d5fef5a0-705a-11e6-9fc5-553c3500b9bf.png">

And the JSON:

```
{
   "type" : "079b1067-c91b-4bec-864b-e88ba6036ddf",
   "element" : {
      "elements" : [
         {
            "label" : "Part 1",
            "type" : "group",
            "elements" : [
               {
                  "reference" : "7501b860-a2ea-46c3-8eed-f0bd806df870",
                  "type" : "input",
                  "label" : "<div>What is your name?</div>"
               },
               {
                  "type" : "input",
                  "label" : "<div>What is your Quest?</div>",
                  "reference" : "7771fbd8-6093-4445-82cb-dbbe3f6de2b1"
               },
               {
                  "inputConfiguration" : {
                     "unitsLabel" : {
                        "rightHandUnitLabel" : "Wind Speed"
                     }
                  },
                  "reference" : "1028e88d-cf23-4515-a174-9d256b630a09",
                  "label" : "<div>What is the windspeed of an unladen Swallow</div>",
                  "type" : "input"
               }
            ],
            "id" : "7d3657c9-ef2d-41f1-ae32-49407cce6f24"
         }
      ],
      "type" : "group",
      "label" : "MCDS Test Form"
   },
   "model" : {
      "uri" : "https://__removed__/model/8/079b1067-c91b-4bec-864b-e88ba6036ddf/1.0",
      "version" : "1.0",
      "properties" : [
         {
            "id" : "7501b860-a2ea-46c3-8eed-f0bd806df870",
            "type" : "string"
         },
         {
            "id" : "1028e88d-cf23-4515-a174-9d256b630a09",
            "type" : "integer"
         },
         {
            "type" : "string",
            "id" : "7771fbd8-6093-4445-82cb-dbbe3f6de2b1"
         }
      ],
      "type" : "079b1067-c91b-4bec-864b-e88ba6036ddf"
   },
   "version" : "1.0",
   "uri" : "https://__removed__/form/8/079b1067-c91b-4bec-864b-e88ba6036ddf/1.0?language=en",
   "language" : "en"
}
```

**Comment by 3lvis** [comment]
Hi @bonebox,

Maybe it's related to ordered relationships. I'm quite busy this week, sorry if I don't have a look earlier. 😞

**Comment by bonebox** [comment]
I understand completely, no worries!

**Comment by 3lvis** [comment]
@bonebox could you please share this in JSON format? Also, this JSON is a list of Forms?

**Comment by 3lvis** [comment]
I'm trying to reproduce this bug :)

**Comment by bonebox** [comment]
Thanks for looking into it!  Here's the JSON.  It is a single form.

```
["model": {
    properties =     (
                {
            id = "7501b860-a2ea-46c3-8eed-f0bd806df870";
            type = string;
        },
                {
            id = "1028e88d-cf23-4515-a174-9d256b630a09";
            type = integer;
        },
                {
            id = "7771fbd8-6093-4445-82cb-dbbe3f6de2b1";
            type = string;
        }
    );
    type = "079b1067-c91b-4bec-864b-e88ba6036ddf";
    uri = "https://_removed_/model/8/079b1067-c91b-4bec-864b-e88ba6036ddf/1.0";
    version = "1.0";
}, "type": 079b1067-c91b-4bec-864b-e88ba6036ddf, "version": 1.0, "uri": https://_removed_/form/8/079b1067-c91b-4bec-864b-e88ba6036ddf/1.0?language=en, "language": en, "element": {
    elements =     (
                {
            elements =             (
                                {
                    label = "<div>What is your name?</div>";
                    reference = "7501b860-a2ea-46c3-8eed-f0bd806df870";
                    type = input;
                    uuid = "727B6E9D-3B03-4731-803B-1894C3C424A5";
                },
                                {
                    label = "<div>What is your Quest?</div>";
                    reference = "7771fbd8-6093-4445-82cb-dbbe3f6de2b1";
                    type = input;
                    uuid = "00FC98C7-9452-4999-82D5-510103C5AD4E";
                },
                                {
                    inputConfiguration =                     {
                        unitsLabel =                         {
                            rightHandUnitLabel = "Wind Speed";
                        };
                    };
                    label = "<div>What is the windspeed of an unladen Swallow</div>";
                    reference = "1028e88d-cf23-4515-a174-9d256b630a09";
                    type = input;
                    uuid = "1BA257C3-FE48-4AD3-8D1D-3BF5FD52AAF2";
                }
            );
            id = "7d3657c9-ef2d-41f1-ae32-49407cce6f24";
            label = "Part 1";
            type = group;
            uuid = "B1EC30A5-E0D5-4DCD-9EAE-5AE8FE190AFC";
        }
    );
    label = "Test Form";
    type = group;
    uuid = "CA057998-CE53-4488-84BA-F53512955FAC";
}]
```

**Comment by bonebox** [comment]
Thanks for looking into it!  Here's the JSON--it's a single form.

```
{
   "type" : "079b1067-c91b-4bec-864b-e88ba6036ddf",
   "element" : {
      "elements" : [
         {
            "label" : "Part 1",
            "type" : "group",
            "elements" : [
               {
                  "reference" : "7501b860-a2ea-46c3-8eed-f0bd806df870",
                  "type" : "input",
                  "label" : "<div>What is your name?</div>"
               },
               {
                  "type" : "input",
                  "label" : "<div>What is your Quest?</div>",
                  "reference" : "7771fbd8-6093-4445-82cb-dbbe3f6de2b1"
               },
               {
                  "inputConfiguration" : {
                     "unitsLabel" : {
                        "rightHandUnitLabel" : "Wind Speed"
                     }
                  },
                  "reference" : "1028e88d-cf23-4515-a174-9d256b630a09",
                  "label" : "<div>What is the windspeed of an unladen Swallow</div>",
                  "type" : "input"
               }
            ],
            "id" : "7d3657c9-ef2d-41f1-ae32-49407cce6f24"
         }
      ],
      "type" : "group",
      "label" : "MCDS Test Form"
   },
   "model" : {
      "uri" : "https://__removed__/model/8/079b1067-c91b-4bec-864b-e88ba6036ddf/1.0",
      "version" : "1.0",
      "properties" : [
         {
            "id" : "7501b860-a2ea-46c3-8eed-f0bd806df870",
            "type" : "string"
         },
         {
            "id" : "1028e88d-cf23-4515-a174-9d256b630a09",
            "type" : "integer"
         },
         {
            "type" : "string",
            "id" : "7771fbd8-6093-4445-82cb-dbbe3f6de2b1"
         }
      ],
      "type" : "079b1067-c91b-4bec-864b-e88ba6036ddf"
   },
   "version" : "1.0",
   "uri" : "https://__removed__/form/8/079b1067-c91b-4bec-864b-e88ba6036ddf/1.0?language=en",
   "language" : "en"
}
```

**Comment by 3lvis** [comment]
Damn, I'm still due to ship my next project and haven't had the time to look at this. I'm sorry if that's causing you troubles.

**Comment by 3lvis** [comment]
@bonebox This should be fixed in the latest release of Sync 1 and Sync 2. Please let me know otherwise.

# [Fixed on Sync 1.14.2](https://github.com/hyperoslo/Sync/releases/tag/1.14.2)

**Comment by bonebox** [comment]
Looking good, thanks @3lvis !
